### PR TITLE
Refactor FXIOS-6775 [v116] Remove old synchronizers

### PR DESCRIPTION
**Pull request by lougeniaC64**
[FXIOS-6775](https://mozilla-hub.atlassian.net/browse/FXIOS-6775)
[15085](https://github.com/mozilla-mobile/firefox-ios/issues/15085)

### Description
This PR is the second of three PRs to cleanup the sync code after the sync manager integration. This PR includes the following:
- Removal of the old synchronizer classes and supporting logic
- Removal of the sync telemetry logic used by the old sync manager
- Removal of the `OldSyncReason` and `OldSyncResult` types and the `MZSyncReason` and `MZSyncResult` type aliases
- Removal of the sync and reset functions defined in the A-S components' storage code for use in the old sync manager
- Removal of related tests

### Pull requests checks where applicable
- [ x ] Fill in the three TODOs above (tickets number and description of your work)
- [ x ] The PR name follows our [naming guidelines](https://github.com/mozilla-mobile/firefox-ios/wiki/Pull-Request-Naming-Guide)
- [ ] Accessibility implemented and tested (minimum Dynamic Text and VoiceOver)
- [ ] Unit tests written and passing
- [ ] Documentation / comments for complex code and public methods

### CODE REVIEW

File: Providers/SyncDisplayState.swift [hunk 1]
```diff
@@ -0,0 +1,47 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
```
Comment (lougeniaC64) [inline Providers/SyncDisplayState.swift:1]: Note: This is a rename of the `SyncStatusResolver.swift` file (which is deleted below). The logic in this file is what remained after all the obsolete sync code was removed. This rename seemed like a better representation of what's left.

**Review by OrlaM — APPROVED** [review]

**Comment by lmarceau** [comment]
@lougeniaC64 could you rebase with main? There was an issue on main that has now been fixed

**Comment by lougeniaC64** [comment]
Huh . . . @lmarceau Is this a Bitrise issue? I haven't removed any metrics so this shouldn't be an issue.

**Comment by lmarceau** [comment]
It was an issue caused by another PR on main, but now should be ok

**Review by nbhasin2 — APPROVED** [review]

**Pull request merged**